### ISSUE TITLE
[nixos] Stage 1 error :  could not insert 'crc32c_intel': No such device

### ISSUE THREAD

**Issue by wizzup** [NixOS/nixpkgs#25734]
## Issue description
* There is error message while booting on stage 1

### Steps to reproduce
* A fresh install from USB boot

## Technical details
* System: NixOS: `17.09pre107265.0afb6d789c (Hummingbird)`
* Nix version: `nix-env (Nix) 1.11.9`
* Nixpkgs version: `"17.09pre107265.0afb6d789c"`
* Sandboxing enabled: `build-use-sandbox = false`

## Additional info
* CPU is `AMD Athlon(tm) II X2 270`

    $ dmesg | grep -i crc32c
    [    0.610578] Btrfs loaded, crc32c=crc32c-generic
    [    0.611970] stage-1-init: loading module crc32c...
    [    0.623988] stage-1-init: modprobe: ERROR: could not insert 'crc32c_intel': No such device

    $ lsmod | grep -i crc32
    crc32c_generic         16384  1

**Comment by 0xABAB** [comment]
@wizzup https://github.com/torvalds/linux/blob/v4.10/arch/x86/crypto/crc32c-intel_glue.c#L4 

Your AMD CPU offers http://www.cpu-world.com/CPUs/K10/AMD-Athlon%20II%20X2%20270%20-%20ADX270OCK23GM%20(ADX270OCGMBOX).html (which does not contain SSE 4.2 support).

The solution is that whatever loads 'crc32c_intel' on your device should have an instruction test of some kind. 

When this is fixed for the AMD code path, it might be fixed for the Atom too.

Which kernel version are you using?

**Comment by wizzup** [comment]
Currently, I am running

    $ uname -a
    Linux earth 4.9.36 #1-NixOS SMP Wed Jul 5 12:40:44 UTC 2017 x86_64 GNU/Linux

I believe it [btrfs](https://github.com/NixOS/nixpkgs/blob/56904d7c423f2b13b37fbd29f39bbb4b52bc7824/nixos/modules/tasks/filesystems/btrfs.nix#L16) that try to load `crc32c` and fallback to `crc32c-generic`

The system, however, doesn't seem to have any noticeable problem other than error message at boot.

**Comment by 0xABAB** [comment]
@wizzup I think the bug is that it is claimed to be an error, while it should be a warning.

**Comment by jfrankenau** [comment]
Yes, Btrfs needs `crc32c` which tries to load `crc32c_intel` if it detects an Intel CPU without checking if this CPU supports SSE 4.2. So, this would probably need fixing upstream and should have been only a warning in first place as it fallbacks to `crc32c_generic` anyway.

**Comment by qweered** [comment]
Not a nixpkgs issue and stale
